llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -119,8 +115,6 @@ RISCVTargetLowering::RISCVTargetLowering(const TargetMachine &TM,
 
   // Set up the register classes.
   addRegisterClass(XLenVT, &RISCV::GPRRegClass);
-  if (Subtarget.is64Bit() && RV64LegalI32)
-    addRegisterClass(MVT::i32, &RISCV::GPRRegClass);
```

Comment:
I guess [this](https://github.com/llvm/llvm-project/commit/c32c0e14d6944d6a0e36191d7b5bf6943f96092e) commit can be reverted as well.

Reply:
I think I have to remove some isel patterns first and those are shared with GISel.